sniffles/test/test_feature.py

```diff
@@ -0,0 +1,310 @@
+from unittest import *
+from sniffles.feature import *
+import random
+
+
+class TestFeature(TestCase):
+    def test_ambigous_notation_class(self):
+        ambigous = AmbiguousNotation("[5,3,8]")
+        self.assertEqual(ambigous.toString(), "[5,3,8]")
+
+    def test_set_notation_class(self):
+        test = SetNotation("{mon,tues,wed,thurs,fri,sat,sun}")
+        testList = ["mon", "tues", "wed", "thurs", "fri", "sat", "sun"]
+        for i in range(0, 10):
+            values = (test.toString()[1:-1]).split(",")
+            for value in values:
+                self.assertTrue(value in testList)
+
+    def test_range_notation_class(self):
+        for i in range(0, 5):
+            lower = random.randint(0, 20)
+            upper = random.randint(21, 25)
+            test = RangeNotation("[" + str(lower) + ":" +
+                                 str(upper) + "]")
+            result = test.toString()
+            myrange = result[1:-1]
+            bounds = myrange.split(":")
+            lower = int(bounds[0])
+            upper = int(bounds[1])
+            self.assertTrue(lower >= 0)
+            self.assertTrue(upper <= 25)
+            self.assertTrue(lower <= upper)
```

Comment:
mylower = random.randint(self.lower_bound, self.upper_bound-1)
        myupper = random.randint(mylower, self.upper_bound)

It can be equal since here is how you implement it in the RangeNotation class. That's why I doubt about that